hook-common/src/pgqueue.rs

```diff
@@ -528,6 +655,7 @@ impl Default for RetryPolicy {
 }
 
 /// A queue implemented on top of a PostgreSQL table.
+#[derive(Clone)]
```

Comment:
`with_state(...)` (see `Router`) appears to be how Axum handles passing non-static state into endpoints. They need to be `Copy`, and cloning `PgPool` is cheap. Cloning the strings seems like a waste but is definitely not going to be on our radar anytime soon... but what we put in here is something we need to keep an eye on.

Unless I'm missing a better way to make this available in my producer endpoints?

Reply:
The pool itself is indeed cheap to clone. We could use a ref count pointer if this struct ever gets expensive to clone.